presto-main/src/main/java/com/facebook/presto/operator/scalar/JsonPathTokenizer.java

```diff
@@ -119,16 +122,46 @@ private String matchQuotedSubscript()
         // quote has already been matched
 
         // seek until we see the close quote
-        int start = index;
-        while (hasNextCharacter() && peekCharacter() != QUOTE) {
+        StringBuilder token = new StringBuilder();
+        boolean escaped = false;
+
+        while (hasNextCharacter() && (escaped || peekCharacter() != QUOTE)) {
+            if (escaped) {
+                switch (peekCharacter()) {
+                    case 'u':
+                        // Unicode escape, will be handled later
```

Comment:
Dear Reviewer,

I am not sure this is right, I could not find any evidence, but this was expected by an existing test: https://github.com/prestodb/presto/blob/c73359fe2173e01140b7d5f102b286e81c1ae4a8/presto-main/src/test/java/com/facebook/presto/operator/scalar/TestJsonExtract.java#L63 . Is the test right?